### ISSUE TITLE
Integrate HA schedule helper to control heating presets 

### ISSUE THREAD

**Issue by Ra72xx** [jmcollin78/versatile_thermostat#122]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Currently, if you want to have time-based switching between heating presets, one way is to

- create a schedule helper, and
- create an automation which uses this schedule to switch thermostat modes.

 I'd like to integrate the automation part into VT, so that you can bind the thermostat to a schedule the same way you can bind it to window sensors. The scheduling then could be turned on/off with a service (or you could integrate the scheduled heating as another preset mode).

**Describe alternatives you've considered**
As described above, you can do all this with a combination of existing HA functionalities, but the workflow would be streamlined if it is coupled with the thermostat. It's basically the same rationale as with integrating window sensor detection in the thermostat.
Other custom integrations like the HACS provided scheduler component/card could be used, but they are also not integrated in the thermostat, are less flexible than the schedule helper (no weekly planning), and only add addtional overhead.

**Additional context**

- IMHO using a schedule helper is the most flexible variant, as you can have different heating programs for every day, and it is integrated in HA with a user-friendly UI.
- Going further, you can also think about switching between different heating schedules from the UI.

**Comment by jmcollin78** [comment]
Hello @Ra72xx ,

Why not using simply the Scheduler and a scheduler card as explained in the readme ? You don't need any automation with this.

I have never tries with a Schedule Helper and I don't see the added value vs the Scheduler integration.

**Comment by Ra72xx** [comment]
I'd like to have defferent heating schedules per day. If you use the custom scheduler component, you have to make different schedules e.g. for weekdays and weekend, in the standard schedule helper this is very easy to achieve with one single weekplan. Moreover also with the scheduler custom components you have to configure your thermostat in different places of the HA UI.
(IMHO for most use cases the standard HA schedule helper can nowadays replace the schedule custom component, it's also easier to use. I have only recently discovered this helper and have now ditched the scheduler component in most cases.)

Basically it's the same with the window sensors - you can also argue that this can be done with a simple automation and needs not to be integrated in the thermostat.

**Comment by jmcollin78** [comment]
I don't know this helpers component. But I will have a look.

**Comment by jmcollin78** [comment]
> Basically it's the same with the window sensors - you can also argue that this can be done with a simple automation and needs not to be integrated in the thermostat.

Not totally. When you integrate, window sensor, presence sensor, energy regulation, ... (all features of VTherm) is it quasi impossible to have this with automations. That is why I do this component.

**Comment by coutadeurf** [comment]
Hi,

I am using a generic automation that works with any scheduler (created thru the helper) and thermostat created by versatile_thermostat:
```
alias: Chauffage
description: ""
trigger:
  - platform: state
    entity_id:
      - schedule.chauffage_salle_de_bain
    id: salle_de_bain
  - platform: state
    entity_id:
      - schedule.chauffage_toilettes
    id: toilettes
  - platform: state
    entity_id:
      - schedule.chambre_parentale
    id: chambre_parentale
  - platform: state
    entity_id:
      - schedule.planning_chauffage_salon
    id: salon
condition:
  - condition: template
    value_template: "{{ trigger.to_state.state in ['on', 'off'] }}"
action:
  - service: climate.set_preset_mode
    data:
      preset_mode: >
        {% if trigger.to_state.state == 'on' %} comfort {% else %} eco {% endif
        %}
    target:
      entity_id: climate.{{ trigger.id }}
mode: parallel
max: 10
```
With this automation, I only have to add the scheduler in the trigger section each time I want to add one and be sure that the ID of the trigger is the same as the name of the thermostat, so for example, my climate entity is climate.salle_de_bain and the trigger id is salle_de_bain.
Not sure it helps but with this approach, no need  to integrate a scheduler in this integration

**Comment by jmcollin78** [comment]
The goal of the integration is to avoid as much as possible the use of Automation so it is a little shame to do that, but it works !

Once again with the Sceduler component, there is no need of Automation and the Scheduler is more flexible than the Helper schedule.
Typically, with Tempo (BLUE / WHITE / RED tarrif from EDF), I need something more complex than just a schedule. The schedule is not the same depending on the color of the day.

**Comment by Ra72xx** [comment]
> Once again with the Sceduler component, there is no need of Automation and the Scheduler is more flexible than the Helper schedule. 

I don't think so, but everybody can do as he pleases. The scheduler helper is IMHO much simpler to set up, and it allows you to create a different schedule for every day of the week with only one helper and with a very usable UI. The only problem is that you need another automation to connect the thermostat with the schedule helper. 
And that was why I thought it would be create to connect a Versatile Thermostat with a schedule helper the same way you connect it with window sensors, to have everything in one place: You could e.g. include the scheduled heating as one of the included preset modes etc.

**Comment by Ra72xx** [comment]
I use this feature request to post my automation, which uses a template helper "schedule.thermostat_heizplan" to switch between eco mode (helper off) and comfort mode (helper on). The automation also respects the window open sensor, as in my case the thermostats always get turned on if the preset is changed, even if the switch for the heater is turned off, so you should have a scheduled change when the window is open. The switch "input_boolean.heizplan_schalter" is used to turn scheduled heating on and off.

```
alias: "Thermostat Heizplan"
description: ""
trigger:
  - platform: state
    entity_id:
      - schedule.thermostat_heizplan
  - platform: state
    entity_id:
      - binary_sensor.thermostat_window_state
    to: "off"
  - platform: state
    entity_id:
      - input_boolean.thermostat_heizplan_schalter
    to: "on"
condition:
  - condition: state
    entity_id: input_boolean.thermostat_heizplan_schalter
    state: "on"
  - condition: state
    entity_id: binary_sensor.thermostat_window_state
    state: "off"
action:
  - choose:
      - conditions:
          - condition: state
            entity_id: schedule.thermostat_heizplan
            state: "off"
        sequence:
          - service: climate.set_preset_mode
            data:
              preset_mode: eco
            target:
              entity_id: climate.smart_thermostat
      - conditions:
          - condition: state
            entity_id: schedule.thermostat_heizplan
            state: "on"
        sequence:
          - service: climate.set_preset_mode
            data:
              preset_mode: comfort
            target:
              entity_id: climate.smart_thermostat
mode: single

```
